src/awkward/operations/ak_from_buffers.py

```diff
@@ -136,6 +136,9 @@ def _impl(
     elif isinstance(form, dict):
         form = ak.forms.from_dict(form)
 
+    if isinstance(length, np.integer):
+        length = int(length)
```

Comment:
If the goal is to ensure that any integer-like `length` is cast as an integer, use the more generic check, `numbers.Integral`.

```suggestion
    if isinstance(length, numbers.Integral):
        length = int(length)
```

(If `numbers` needs to be imported to do this, then that change is needed as well.) I could have added `and not isinstance(length, int)`, but the `int` function is idempotent.

By the way, what does this change have to do with the original issue? I'm seeing a few irrelevant diffs, like we're not looking at the difference between your branch and `main`, but that's what GitHub says it's comparing.

In fact, this is in `main`:

https://github.com/scikit-hep/awkward/blob/39541dbf38452d1e0a3dbf2a3097e2dbd68b3521/src/awkward/operations/ak_from_buffers.py#L139-L140

Ah: @maxgalli added it earlier today and @agoose77 approved and merged it. If I had looked at it closely, I personally would have made the above change, and we can still make it, but it would be a new PR.

More importantly, it looks like GitHub is showing differences with respect to the time when @pfackeldey forked the repo. But `wants to merge 9 commits into scikit-hep:main from pfackeldey:fix/issue_3118_prevent_exponential_memory_growth_in_unionarray` is supposed to show diffs with respect to the current `scikit-hep:main`. I think we're looking at a GitHub bug.

Reply:
Sorry, I messed something up syncing my fork with the latest additions in awkward by you and @maxgalli . It should be solved now.